quadratichq/quadratic
# feat: show active cell selection formula results in bottom bar

## Background

Sheets and Excel both have a feature that runs six formulas (like sum and average) for an active multi-cell selection. In this way, you don't have to write a formula to see its output. You can just make a selection on the sheet and see the output change as you change your selection.

### Formulas

1. `AVERAGE`
2. `COUNTA`
3. Numerical count
4. `MIN`
5. `MAX`
6. `SUM`

### Screenshot examples

Excel web:

<img width="467" alt="CleanShot 2023-07-03 at 22 56 10@2x" src="https://github.com/quadratichq/quadratic/assets/1316441/a6518248-3406-4b4e-9441-9f89056c4767">

Excel native:

<img width="475" alt="CleanShot 2023-07-03 at 22 51 32@2x" src="https://github.com/quadratichq/quadratic/assets/1316441/c0d00667-afa1-48c5-8273-ad9ec9a6b065">

Sheets:

<img width="96" alt="CleanShot 2023-07-03 at 22 57 06@2x" src="https://github.com/quadratichq/quadratic/assets/1316441/4e6076e3-38eb-4cbe-b5cb-8c49cea7b794">
<img width="202" alt="CleanShot 2023-07-03 at 22 57 14@2x" src="https://github.com/quadratichq/quadratic/assets/1316441/2a544b71-e807-41f7-8b64-b35ccce1ecac">

## Quadratic

Excel and Sheets implement this feature slightly differently, but the main idea is that the UI becomes configureable and you can choose which formula you want to see in the bottom bar (sheets only ever allows once choice, while excel is configureable and lets you choose which of the 6 you want to see).

To start in Quadratic, we'll provide Sum, Average, and Count. We can add more later as we deem necessary.

Working example:

![Jul-03-2023 23-04-34](https://github.com/quadratichq/quadratic/assets/1316441/a8781ce9-b0db-412a-a558-39f6564c9559)

### To Test

Make various selections across the grid and ensure the following:

- Only show `count` if at least two or more cells with data are selected.
- Only show `sum` and `average` if at least one or more cells of the active selection are numbers.
